frontend/app/controllers/spree/products_controller.rb

```diff
@@ -18,15 +18,11 @@ def show
       @variants = @product.variants_including_master.active(current_currency).includes([:option_values, :images])
       @product_properties = @product.product_properties.includes(:property)
       @taxon = Spree::Taxon.find(params[:taxon_id]) if params[:taxon_id]
-      # If an old id or a numeric id was used to find the record, then
-      # the request path will not match the product_path, and we should do
-      # a 301 redirect that uses the current friendly id.
-      if request.path != spree.product_path(@product)
-        return redirect_to @product, status: :moved_permanently
-      end
+      redirect_legacy_paths
```

Comment:
This should be within the conditional still though, which is probably why there are failing specs. You don't always redirect just only when it's legacy.